File: robocode-tankroyale-bot-api-csharp/src/events/BotDeathEvent.cs
using Newtonsoft.Json;

namespace Robocode.TankRoyale.BotApi.Events
{
  /// <summary>
  /// Event occurring when a bot has died.
  /// </summary>
  public sealed class BotDeathEvent : Event
  {
    /// <summary>The id of the bot that has died.</summary>
    public int VictimId { get; }

    /// <summary>Initializes a new instance of the BotDeathEvent class.</summary>
    /// <param name="turnNumber">Turn number.</param>
    /// <param name="victimId">Id of the bot that has died.</param>
    [JsonConstructor]
    public BotDeathEvent(int turnNumber, int victimId) : base(turnNumber) => VictimId = victimId;
  }
}

File: robocode-tankroyale-server-kotlin/src/main/kotlin/dev/robocode/tankroyale/server/event/BotDeathEvent.kt
package dev.robocode.tankroyale.server.event

/** Event sent when a bot has been killed. */
data class BotDeathEvent(
    /** Turn number when event occurred */
    override val turnNumber: Int,

    /** Bot id of the victim that got killed */
    val victimId: BotId,

) : Event()
